dev/ansible/04_test_persistence.yml

```diff
@@ -0,0 +1,182 @@
+# This file contains checks and test scenarios that test the persistence feature for postgres
+- name: set variable
+  set_fact:
+    consul_token: "master"
+  tags: test
+
+- name: Fetch bootstrap token nomad
+  shell:
+    cmd: vault kv get -field=secret-id secret/nomad-bootstrap-token
+  when: lookup('env', 'nomad_acl') | bool
+  register: bootstrap_token
+  tags: test
+
+- name: Verify if host volume has been added on nomad node(acl enabled)
+  shell:
+    cmd: nomad node status -short -self | grep 'Host Volumes' | tr -d " \t\n\r"  | cut -d "=" -f2
+  register: nomad_node_status
+  when: lookup('env', 'nomad_acl') | bool
+  failed_when: nomad_node_status.stdout != "persistence"
+  environment:
+    NOMAD_TOKEN: "{{ bootstrap_token.stdout }}"
+  tags: test
+
+- name: Verify if host volume has been added on nomad node(acl disabled)
+  shell:
+    cmd: nomad node status -short -self | grep 'Host Volumes' | tr -d " \t\n\r"  | cut -d "=" -f2
+  register: nomad_node_status_1
+  failed_when: nomad_node_status_1.stdout != "persistence"
+  when: not lookup('env', 'nomad_acl') | bool
+  tags: test
+
+#******************************************************************#
+#Test Scenario 1
+#******************************************************************#
+# 1. Stop the postgres job
+# 2. Check status for postgres and it should not be in running state
+# 3. Delete terraform.tfstate, terraform.tfstate.backup and .terraform
+# 4. Re-deploy postgres with existing volume using terraform apply
+# 5. Verify that data persists on new instance
+# 6. Verify that it is possible to upload data to the new instance
+# https://unix.stackexchange.com/a/148698
+
+- name: Start local proxy on port 9999 to postgres service
+  shell: nohup consul connect proxy -service postgres-local -upstream postgres:9999 -log-level debug </dev/null >/dev/null 2>&1 &
```

Comment:
If you use vagrant-hashistack 0.4.3 once consul_acl enabled, you have to also create intentions for service `postgres-local`

Reply:
Also related tasks https://github.com/fredrikhgrelland/terraform-nomad-postgres/issues/22